tools/js-sdk-release-tools/src/autoGenerateInPipeline.ts

```diff
@@ -32,13 +35,25 @@ async function automationGenerateInPipeline(
         skipGeneration,
         runningEnvironment,
         typespecProject,
-        autorestConfig
+        autorestConfig,
+        apiVersion,
+        sdkReleaseType,
     } = await parseInputJson(inputJson);
 
+    const currAPIVersion = inputApiVersion ?? apiVersion;
```

Comment:
`apiVersion` should only be overridden when another parameter `runMode` equals 'release' or 'local'. Refer to [comment](https://github.com/Azure/azure-sdk-tools/issues/10155#issuecomment-2873004064) for details.